apps/prairielearn/src/lib/json.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Given an original object and a modified object, returns a new object
+ * that contains all the data from the modified object, but with the keys
+ * matching the order that they appear in the original object. Keys that are
+ * not present in the original object are added at the end.
+ *
+ *
+ * This function is meant to be used when programmatically editing JSON data.
+ * It's designed to minimize the changes in the JSON data such that a `git diff`
+ * doesn't show spurious changes unrelated to the actual data changes.
+ */
+export function applyKeyOrder(original: any, modified: any): any {
```

Comment:
I'd be happy to bikeshed on alternative names.

Reply:
I think `applyKeyOrder` is an apt name. Some additional suggestions could be something like `alignJsonKeys` or `reorderModifiedKeys`. It's a tough function to name, I think  `applyKeyOrder` works fine here.